src/sinks/http.rs

```diff
@@ -227,8 +229,13 @@ impl HttpSink for HttpSinkConfig {
             .uri(uri)
             .header("Content-Type", ct);
 
-        if let Some(ce) = self.compression.content_encoding() {
-            builder = builder.header("Content-Encoding", ce);
+        if let Compression::Gzip(level) = self.compression {
```

Comment:
Could we make this a `match` to ensure we handle any additional compression formats we add in the future?

Reply:
I think `clippy` will throw errors for match with 1 arm..

Reply:
It seems like it could have at least two arms though, no? `Compression::Gzip(level)` and `None`.

Reply:
```rust
        match self.compression {
            Compression::Gzip(level) => {
                builder = builder.header("Content-Encoding", "gzip");

                let level = level.unwrap_or(GZIP_DEFAULT) as u32;
                let mut w = GzEncoder::new(Vec::new(), flate2::Compression::new(level));
                w.write_all(&body).expect("Writing to Vec can't fail");
                body = w.finish().expect("Writing to Vec can't fail");
            }
            Compression::None => {}
        }
```
works for clippy :+1:

Reply:
:+1: that would assuage my concern that we forget to update this if we add another supported compression format.